### PR TITLE
Adding AWS tags

### PR DESCRIPTION
**Pull request by gaspardmoindrot**

### CODE REVIEW

File: srcs/runners_manager/vm_creation/aws/schema.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 from marshmallow import Schema, fields
 
+class AwsCloudConfig(Schema):
+    aws_tags = fields.Dict(keys=fields.Str(), values=fields.Str())
```
Comment (tcarmet) [inline srcs/runners_manager/vm_creation/aws/schema.py:4]: Seems a bit repetitive to mention `aws` in a config that is specific to `aws`. I wouldn't mind the config to look like this at the end:

```yaml
cloud_name: aws
cloud_config:
  tags:
    my_tag: value
```

```suggestion
    tags = fields.Dict(keys=fields.Str(), values=fields.Str())
```
If you want to differentiate `tags` of github runners and aws tags you can use `ec2_tags` that's a legitimate specification.

And please ensure the value is set as optional (most likely is already the case, just want to double check)

**Review by tcarmet — COMMENTED** [review]
You're heading in the right direction so the code as is LGTM.

Couple of stuff missing though:

- [ ] Update the doc accordingly to mention the tags availability.
- [ ] Ensure the helm chart is compatible with the new config (I think it is already but just double check).

File: srcs/runners_manager/vm_creation/aws/schema.py (hunk 1 above)
Comment (gaspardmoindrot) [reply srcs/runners_manager/vm_creation/aws/schema.py:4]: I check and change everything once I get it to work! Thanks for the comment!

Comment (gaspardmoindrot) [reply srcs/runners_manager/vm_creation/aws/schema.py]: Done, everything's good! It is optional, I just checked

File: srcs/runners_manager/vm_creation/aws/AwsManager.py [hunk 2]
```diff
@@ -27,7 +29,8 @@ def __init__(
         super(AwsManager, self).__init__(
             name, settings, redhat_username, redhat_password, ssh_keys
         )
-        self.ec2 = boto3.client("ec2")
+        self.tags = settings.get("tags")
+        self.ec2 = boto3.client("ec2", region_name='us-west-2')
```
Comment (tcarmet) [inline srcs/runners_manager/vm_creation/aws/AwsManager.py:33]: We were using the `AWS_DEFAULT_REGION` environment var to set this setting. What happened?

We cannot hardcode the value in the code.
```suggestion
        self.ec2 = boto3.client("ec2")
```

**Review by tcarmet — CHANGES_REQUESTED** [review]

File: srcs/runners_manager/vm_creation/aws/AwsManager.py (hunk 2 above)
Comment (gaspardmoindrot) [reply srcs/runners_manager/vm_creation/aws/AwsManager.py:33]: Damn I forgot to remove it or to do it properly, but indeed, I don't know why but if you don't give this parameter in addition it returns an error. I'll look for a way to fix it (or make it cleaner)

Comment (gaspardmoindrot) [reply srcs/runners_manager/vm_creation/aws/AwsManager.py:33]: Here is the error : `botocore.exceptions.NoRegionError: You must specify a region.`

Comment (gaspardmoindrot) [reply srcs/runners_manager/vm_creation/aws/AwsManager.py:33]: Ok, it's good now

**Review by tcarmet — APPROVED** [review]

**Pull request merged**